haskell/private/runghc.bzl

```
@@ -0,0 +1,64 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+# Note [Running Setup.hs when cross-compiling]
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# When cross compiling we interpret Setup.hs in the execution
+# platform. In order to achieve this, we provide a runghc of a
+# toolchain targeting the execution platform.
+#
+# Producing a runghc wrapper with a custom rule, as defined here,
+# allows to use it in rules which are using a Haskell toolchain for
+# the target platform.
+#
+
+def _runghc_wrapper_impl(ctx):
+    hs_toolchain = ctx.toolchains["@rules_haskell//haskell:toolchain"]
+
+    f = hs_toolchain.tools.runghc
+    runghc_runfile_path = paths.join(f.owner.workspace_name, f.owner.package, f.owner.name)
+    runghc_wrapper_file = ctx.actions.declare_file(ctx.label.name)
+    ctx.actions.write(
+        output = runghc_wrapper_file,
+        content = """\
+#!/usr/bin/env python3
+
+import subprocess
+import sys
+from rules_python.python.runfiles import runfiles
+
+r = runfiles.Create()
+
+subprocess.run([r.Rlocation("{runghc}")] + sys.argv[1:], check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Letting the subprocess use stdout and stderr from the caller:
```suggestion
subprocess.run([r.Rlocation("{runghc}")] + sys.argv[1:], check=True)
```